mlir/test/Dialect/Linalg/invalid.mlir

```diff
@@ -455,6 +455,18 @@ func.func @map_input_output_shape_mismatch(
 
 // -----
 
+func.func @map_no_operands(
+    %lhs: tensor<64xf32>, %rhs: tensor<64xf32>, %init: tensor<64xf32>)
+    -> tensor<64xf32> {
+  // This must not crash the parser.
+  linalg.map { arith.addf }
```

Comment:
Why is this op not rejected by the verifier? `linalg.map` has a region according to the TableGen file:
```
let regions = (region SizedRegion<1>:$mapper);
```
But your new implementation never calls `addBodyWithPayloadOp`, so no region is added to the op. Am I missing something?
```c++
  if (payloadOpName.has_value()) {
    if (!result.operands.empty())
      addBodyWithPayloadOp(parser, result, payloadOpName.value(),
                           payloadOpAttrs,
                           ArrayRef(result.operands).drop_back());
  }
```

Reply:
Also, the `init` operand is not optional, yet another reason why this should fail verification.
```
    // Output arg
    TensorOrMemref:$init
```

Reply:
You're right, I got confused here. But the verifier gets tripped correctly when putting the Op in a separate function. When both versions (with and without result) are in the same function, the verifier always (?) trips on the Op with a result. Do you have an idea why this is?

Reply:
Maybe it stops verifying after the first error. Also the `cannot name an operation with no results` is not an op verifier error. It may not even get to the op verification phase.

Reply:
Ah yes. Those cases it errors out in the final sanity checking at the end of the Parser and never get to the verifier, so I never should have put both tests into the same `FuncOp`.